### ISSUE TITLE
[Fix] Artifacts/Frozen first frame in vid2vid (HuMo) caused by latent padding logic

### ISSUE THREAD

**Issue by smithian123** [kijai/ComfyUI-WanVideoWrapper#1909]
I encountered an issue when using the Wan-based HuMo model where the generated video would often have a "glitched" or static first frame.
The Issue:
In nodes_sampler.py, the current vid2vid logic handles mismatched latent lengths (where input < target) by repeating the first frame and prepending it to the start of the sequence.
input_samples = torch.cat([input_samples[:, :1].repeat(...), input_samples], dim=1)
This breaks motion continuity at the very beginning of the generation, resulting in a visual artifact or freeze on frame 0.
The Fix:
I modified the logic to perform Tail Padding instead of Head Padding.
If the input is too short, it repeats the last frame and appends it to the end.
If the input is too long, it crops from the end.
This resolves the first-frame glitch and maintains smooth motion initiation.
Proposed Change
File: nodes_sampler.py
Location: Inside the process method (approx line 2640 in current main branch), specifically the vid2vid block.
Replace this original block:
code
Python
input_samples = samples["samples"].squeeze(0).to(noise)
            if input_samples.shape[1] != noise.shape[1]:
               input_samples = torch.cat([input_samples[:, :1].repeat(1, noise.shape[1] - input_samples.shape[1], 1, 1), input_samples], dim=1)
With this updated logic:
code
Python
input_samples = samples.get("samples", None)
            if input_samples is not None:
                input_samples = input_samples.squeeze(0).to(noise)

                # --- FIX: SAFE TAIL-PAD / TAIL-CROP ---
                if input_samples.shape[1] != noise.shape[1]:
                    in_T = input_samples.shape[1]
                    out_T = noise.shape[1]

                    if in_T < out_T:
                        # Pad at the END with the last latent frame to prevent start-glitches
                        pad = input_samples[:, -1:].repeat(1, out_T - in_T, 1, 1)
                        input_samples = torch.cat([input_samples, pad], dim=1)
                    else:
                        # Crop extra frames from the END
                        input_samples = input_samples[:, :out_T]
                # --------------------------------------
Tested on Wan2.1 i2v workflows with HuMo, confirming the startup artifacts are resolved.
